spaceR/spaceR/Controllers/RocketsControlller.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  RocketsControlller.swift
+//  spaceR
+//
+//  Created by Kieran Woodrow on 2022/02/25.
+//
+
+import Foundation
+
+
+import UIKit
+
+class RocketsController: UIViewController {
+    @IBOutlet weak var tableView: UITableView!
+    //Creates an array of rockets.
+    private var rockets: [Rocket] = []
+    //private var launches: [Launch] = []
+   
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        //Calls the function that will perform API call for all rockets
+        getAllRockets()
+        //getAllLaunches()
+        setupTableView()
+        print(rockets)
+    }
+    
+    private func setupTableView() {
+        self.tableView.delegate = self
+        self.tableView.dataSource = self
+    }
+    
+    //Function that will process the population of the rockets variable
+    func getAllRockets(){
+        //Use heper function from Endpoint.swift file
+        URLSession.shared.getAllRocketsEndpointURL(url: Constants.getAllRocketsUrl, model: [Rocket].self){ result in
+            //Swift the state of the result
+            switch result{
+            case .success(let userArray):
+                //set rockets array to userarray
+                self.rockets = userArray
+                //Recall tablevview to reload data
+                self.tableView.reloadData()
+               // print(userArray)
+                print("--------ALL AVAILABLE ROCKET NAMES-------------------")
```

Comment:
You can also get rid of these print statements 😄 we don't really want to commit our code we used for debugging.

Reply:
will do